File: packages/core/src/lib/link/versioned-links.ts
import { CaseStyle } from "kryo";
import { RecordIoType, RecordType } from "kryo/lib/record.js";

import { $VersionedDinoparcLink, VersionedDinoparcLink } from "./versioned-dinoparc-link.js";
import { $VersionedHammerfestLink, VersionedHammerfestLink } from "./versioned-hammerfest-link.js";
import { $VersionedTwinoidLink, VersionedTwinoidLink } from "./versioned-twinoid-link.js";

/**
 * Versioned links from an Eternal-Twin user.
 */
export interface VersionedLinks {
  dinoparcCom: VersionedDinoparcLink;
  enDinoparcCom: VersionedDinoparcLink;
  hammerfestEs: VersionedHammerfestLink;
  hammerfestFr: VersionedHammerfestLink;
  hfestNet: VersionedHammerfestLink;
  spDinoparcCom: VersionedDinoparcLink;
  twinoid: VersionedTwinoidLink;
}

export const $VersionedLinks: RecordIoType<VersionedLinks> = new RecordType<VersionedLinks>({
  properties: {
    dinoparcCom: {type: $VersionedDinoparcLink},
    enDinoparcCom: {type: $VersionedDinoparcLink},
    hammerfestEs: {type: $VersionedHammerfestLink},
    hammerfestFr: {type: $VersionedHammerfestLink},
    hfestNet: {type: $VersionedHammerfestLink},
    spDinoparcCom: {type: $VersionedDinoparcLink},
    twinoid: {type: $VersionedTwinoidLink},
  },
  changeCase: CaseStyle.SnakeCase,
});

File: clients/kotlin/src/main/kotlin/net/eternaltwin/link/VersionedLinks.kt
package net.eternaltwin.link

import kotlinx.serialization.*

@Serializable
data class VersionedLinks constructor(
  @SerialName("dinoparc_com")
  val dinoparcCom: VersionedDinoparcLink,
  @SerialName("en_dinoparc_com")
  val enDinoparcCom: VersionedDinoparcLink,
  @SerialName("hammerfest_es")
  val hammerfestEs: VersionedHammerfestLink,
  @SerialName("hammerfest_fr")
  val hammerfestFr: VersionedHammerfestLink,
  @SerialName("hfest_net")
  val hfestNet: VersionedHammerfestLink,
  @SerialName("sp_dinoparc_com")
  val spDinoparcCom: VersionedDinoparcLink,
  val twinoid: VersionedTwinoidLink,
)
